vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -27,7 +27,9 @@ def openstack_handle(options = {})
       raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
       username = options[:user] || self.authentication_userid(options[:auth_type])
-      password = options[:pass] || self.authentication_password(options[:auth_type])
+
+      # the |"" is here to prevent an unhelpful ArgumentError from OpenStackHandle::Handle#connect
+      password = options[:pass] || self.authentication_password(options[:auth_type]) || ""
```

Comment:
actually, @blomquisg I was thinking that the interface of authentication mixin is the class should have a method with a list of required_credential_fields

```ruby
#authentication mxin
Module AuthenticationMixin
  def self.required_credential_fields?
    [:userid]
  end

  def has_credentials?(type = nil)
    required_credential_fields.all?{ |field|  !!authentication_component(type, field) }
  end

  def missing_credentials?(type = nil)
    !has_credentials?(type)
  end

...
end



# some class
class SomeProvider
  include AuthenticationMixin

  def required_credential_fields
    [:userid, :password]
  end
end

```